### ISSUE TITLE
Failed "creating native project directories" does not cause Prebuild phase to fail

### ISSUE THREAD

**Issue by bartvanandel** [expo/eas-cli#1540]
### Summary

Builds fired up using `eas build` fail due to a credentials problem.
_NB: not sure if this is the right repo for the issue, but the failing command seems to be part of this one._

While debugging the issue (we're using a private Azure Artifacts repository), I noticed that Expo thinks the Prebuild step succeeded, while clearly the logs indicate that in fact it could not proceed due to a credentials problem, as the following screenshot shows:

![image](https://user-images.githubusercontent.com/1194804/201906337-26f03687-8a53-4265-8d1a-4bc03d2232a0.png)

Installing dependencies from the same artifacts repository did in fact work 3 steps earlier in the same build, so the npm setup that we're using seems to be at least partially correct.

The build finally fails with the following messages:

![image](https://user-images.githubusercontent.com/1194804/201906663-02cb909e-0743-4c8d-aef5-47bbac9e33b2.png)

### Environment

    expo-env-info 1.0.5 environment info:
      System:
        OS: Windows 10 10.0.22000
      Binaries:
        Node: 16.14.0 - C:\ProgramData\scoop\apps\nvm\current\nodejs\nodejs\node.EXE
        npm: 8.3.1 - C:\ProgramData\scoop\apps\nvm\current\nodejs\nodejs\npm.CMD
      npmPackages:
        expo: ~47.0.3 => 47.0.3
        react: 18.1.0 => 18.1.0
        react-dom: 18.1.0 => 18.1.0
        react-native: 0.70.5 => 0.70.5
        react-native-web: ~0.18.7 => 0.18.10
      Expo Workflow: managed

### Please specify your device/emulator/simulator platform, model and version

N/A

### Error output

See screenshots

### Reproducible demo or steps to reproduce from a blank project

Will try to provide if the above is not enough info

**Comment by bartvanandel** [comment]
For the one about the failed script in the Pre-install hook, I can: 6ee36d19-fcff-48b7-9601-de22a439eeba

For the Prebuild phase that should have failed, I'm afraid I deleted the build. But let me see if I can create a quick reproduction repo.

**Comment by bartvanandel** [comment]
I'm struggling to reproduce this issue myself. I know it's only 10 days ago, but we've been moving so fast I can't quite remember. It was something like the following, however with an actually existing Azure artifacts repo, that somehow succeeded to install during the "Install dependencies" phase but failed later during the "Prebuild" phase. The actual error that popped up can be seen in the screenshot provided earlier. I was trying to fight the wrong problem though because the Expo build logs didn't show that step as having failed.

As I mentioned in my other comment, other steps also don't always fail if the script that is being run by Expo (or EAS) fails. Looks like the result code from the child processes isn't always taken into account properly. This should not be too difficult to find in your source code. I'm afraid though that the logs of our failing build have been deleted, and the intermittent setup that caused the failing build wasn't saved either (not even in the git reflog; I've only committed the working configuration).

Steps taken to try and reproduce, although I don't think they're particularly useful in this case because they fail properly, as they should:
- `npx create-expo-app issue-1540`
- `cd issue-1540`
- `eas init`
  - choose my account: "bartvanandel_dytter"
- create "eas.json" from the example here: https://docs.expo.dev/build/eas-json/
- create ".npmrc" with the following content:
  ```
  registry=https://pkgs.dev.azure.com/nonexistent/_packaging/nonexistent/npm/registry/
  always-auth=true
  ```
- run `eas build --profile preview --platform android`, accepting defaults on the questions

This build failed during installing dependencies, as would be expected. Build details here: https://expo.dev/accounts/bartvanandel_dytter/projects/issue-1540/builds/b6d02955-111e-4848-9dca-fc5b9cc885cc

- changed the contents of ".npmrc" to the following:
  ```
  ; begin auth token
  //pkgs.dev.azure.com/nonexistent/_packaging/nonexistent/npm/registry/:username=nonexistent 
  //pkgs.dev.azure.com/nonexistent/_packaging/nonexistent/npm/registry/:_password=invalid_password
  //pkgs.dev.azure.com/nonexistent/_packaging/nonexistent/npm/registry/:email=npm requires email to be set but doesn't use the value
  //pkgs.dev.azure.com/nonexistent/_packaging/nonexistent/npm/:username=nonexistent 
  //pkgs.dev.azure.com/nonexistent/_packaging/nonexistent/npm/:_password=invalid_password
  //pkgs.dev.azure.com/nonexistent/_packaging/nonexistent/npm/:email=npm requires email to be set but doesn't use the value
  ; end auth token
  
  registry=https://pkgs.dev.azure.com/nonexistent/_packaging/nonexistent/npm/registry/
  always-auth=true
  ```
- ran the build command again

This build failed in the same way. In both cases, it's caused by a 401 (which is what Azure returns). Details for this build: https://expo.dev/accounts/bartvanandel_dytter/projects/issue-1540/builds/6a472541-b9a0-4a6d-a4ae-f974e772c8e6

**Comment by dsokal** [comment]
I don't see how it's a problem with EAS Build. You're saying you can't repro this so I'm not sure what we need to fix.
If you come across a failure that you think is a bug with EAS Build, please open another issue with repro steps.

**Comment by bartvanandel** [comment]
I was a bit too eager cleaning up the failed Expo builds for sure. For the record: I added this bug in the Expo repo because that's where I saw the code that produced the error that I encountered. It got moved here.

I do think that build steps should not show "success" if the underlying scripts failed, which is why I've posted this bug. It's not just the step that I mentioned initially, see also comments added below the initial one. But if you don't want to pursue this without a set of repro steps, fine. If I run into it again, I know what to do.
